posts/redux-sagas-all-you-need-to-know.md

```diff
@@ -0,0 +1,107 @@
+# <img src="https://redux-saga.js.org/img/Redux-Saga-Logo.png" width="50"> Redux Sagas: All You Need to Know
+In the world of React web development, managing state is an essential task. Redux is a popular state management library used by many to store and manage the state of the application.
+
+However, when it comes to handling complex async logic, Redux on its own isn't always enough. This is where Redux Sagas come in, providing a solution to deal with async logic in a more declarative way.
+
+In this blog post, we'll take a deep dive into Redux Sagas, exploring what they are, why we would use them, their advantages over other libraries, and some example scenarios to help better understand their implementation.
+ 
+## What are Redux Sagas?
+Redux Sagas have gained popularity among developers lately as a middleware library used to handle complex side effects (e.g. async operations such as API calls, timers, and socket connections) in an elegant way by using ES6 generators instead of traditional callbacks or promises. This allows you to pause and resume the function at any point, making the flow of data more easily understood and predictable.
+
+It also helps to decouple the business logic of handling asynchronous operations and state management. Using Sagas can help you to write cleaner and more maintainable code. It can also provide better error handling and more advanced testing capabilities, as you can test the Sagas' logic directly.
+
+They are designed to be run in parallel with your Redux reducers, which means they don't block the main thread and can be used in conjunction with other Redux middleware (e.g., Redux Thunk).
+ 
+## Why use Redux Sagas?
+One of the most significant advantages of Redux Sagas over other middleware like is their handling of side effects natively. As we know, side effects are the operations that affect our application's state outside the current scope of a function.
+
+Sagas solve this problem by providing a simple way to manage and control these side effects. They also provide a centralised location for all your side effects code, which enables you to separate complex async logic from our components, making them more reusable. Sagas provide a more robust way to manage long-running processes such as WebSocket connections or polling APIs.
+
+They also provide an action dispatch flow that allows to pause, delay and cancel promises, making it easier to handle race conditions, manage errors and provide more control over side effects.
+
+Another advantage of using Redux Sagas is that they allow us to test our code more efficiently. Since Sagas manage async in a very isolated way, you can test them just like you test your reducers, which makes your app more robust over time.
+ 
+## Comparison to other state management
```

Comment:
I would probably move this entire section under the Advantages/Disadvantages of Sagas, as a counterpoint to the disadvantages.